testsuite/MDAnalysisTests/core/test_topologyobjects.py

```diff
@@ -430,3 +504,186 @@ def test_TG_getitem_bool_IE(self):
         tg = self.universe.atoms.bonds[10:13]
         tg2 = tg[sel]
         assert_equal(len(tg2), 0)
+
+    # atomX access
+    def test_atom1(self):
+        tg = self.universe.bonds[:5]
+
+        a1 = tg.atom1
+        assert_(len(tg) == len(a1))
+        for (atom, bond) in zip(a1, tg):
+            assert_(atom == bond[0])
+
+    def test_atom2(self):
+        tg = self.universe.bonds[:5]
+
+        a2 = tg.atom2
+        assert_(len(tg) == len(a2))
+        for (atom, bond) in zip(a2, tg):
+            assert_(atom == bond[1])
+
+    def test_atom3_IE(self):
+        tg = self.universe.bonds[:5]
+
+        assert_raises(IndexError, getattr, tg, 'atom3')
+
+    def test_atom3(self):
+        tg = self.universe.angles[:5]
+
+        a3 = tg.atom3
+        assert_(len(tg) == len(a3))
+        for (atom, bond) in zip(a3, tg):
+            assert_(atom == bond[2])
+
+    def test_atom4_IE(self):
+        tg = self.universe.bonds[:5]
+
+        assert_raises(IndexError, getattr, tg, 'atom4')
+
+    def test_atom4(self):
+        tg = self.universe.dihedrals[:5]
+
+        a4 = tg.atom4
+        assert_(len(tg) == len(a4))
+        for (atom, bond) in zip(a4, tg):
+            assert_(atom == bond[3])
+
+
+class TestTopologyGroup_Cython(object):
```

Comment:
Just the raw maths on calculating distances/angles eg: https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/lib/include/calc_distances.h#L334